File: thyoondev/콜라츠 추측.js
function solution(num) {
  let count = 0;
  let n = num;

  while (n !== 1 && count < 500) {
    n = n % 2 == 0 ? n / 2 : n * 3 + 1;
    count++;
  }

  return n === 1 ? count : -1;
}

File: simchodi/콜라츠추측.java
class Solution {
    public int solution(int num) {
        int count = 0;
        long num2 = (long)num;
        while (num2 != 1){
            if (count==499){
                break;
            }
            if (num2 % 2 ==0){
                num2 = num2/2;
            }else{
                num2 = (num2 * 3) + 1;
            }
            count++;
        }
        return count==499?-1:count;
    }

}
